debug/ftrace.rst

```diff
@@ -0,0 +1,36 @@
+.. _ftrace:
+
+Ftrace
+######
+This section describes how to generate function call graph for user Trusted
+Applications using ``ftrace``.
+
+The configuration option ``CFG_TA_FTRACE_SUPPORT=y`` enables OP-TEE to collect
+function graph information from Trusted Applications running in user mode and
+compiled with ``-pg``. Once collected, the function graph data is formatted
+in the ``ftrace.out`` format and sent to ``tee-supplicant`` via RPC, so they
+can be saved to disk, later processed and displayed using helper script called
+``symbolize.py`` present as part of ``optee_os`` repo.
+
+Usage
+*****
+
+    - Build OP-TEE OS and OP-TEE Client with ``CFG_TA_FTRACE_SUPPORT=y``. You
+      may also set ``CFG_ULIBS_INSTRUMENT=y`` in OP-TEE OS to instrument the
+      user TA libraries (libutee, libutils, libmpa).
+
+    - Build user TAs with ``-pg``, for instance using: ``CFLAGS_ta_arm64=-pg``.
+      Note that instrumented TAs have a larger ``.bss`` section. The memory
+      overhead depends on FTRACE_BUF_SIZE macro which can be configured
```

Comment:
> s/CFG_ULIBS_INTRUMENT/\``CFG_ULIBS_INTRUMENT\``/

Sorry, bad copy/paste. I meant s/FTRACE_BUF_SIZE/\`\`FTRACE_BUF_SIZE\`\`/:
Add \`\` around `FTRACE_BUF_SIZE` for formatting purpose,  _(not around `CFG_ULIBS_INTRUMENT` :()_

Reply:
Ok, will add `` around FTRACE_BUF_SIZE for formatting purpose.